src/Surfnet/StepupMiddleware/CommandHandlingBundle/DependencyInjection/CompilerPass/AddPipelineStagesCompilerPass.php

```diff
@@ -18,19 +18,45 @@
 
 namespace Surfnet\StepupMiddleware\CommandHandlingBundle\DependencyInjection\CompilerPass;
 
+use Symfony\Component\Config\Definition\Exception\InvalidConfigurationException;
 use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Exception\RuntimeException;
 use Symfony\Component\DependencyInjection\Reference;
 
 class AddPipelineStagesCompilerPass implements CompilerPassInterface
 {
+    /**
+     * {@inheritdoc} Since the priorities cannot be changed runtime but only through configuration, we're doing the
+     * sorting based on priority here. A lower priority means the stage is added earlier.
```

Comment:
Yep, there was no mechanism to force priority other than order of declaration in the `pipeline.yml` file, which I found out when appending :D